fs/dafsclient/rmtclient.cpp

```diff
@@ -431,29 +436,28 @@ void receiveDaFsBuffer(ISocket * socket, MemoryBuffer & tgt, unsigned numtries,
             size32_t szread;
             SOCKREADTMS(socket)((gotLength<4000)?tgt.reserve(gotLength):tgt.reserveTruncate(gotLength), gotLength, gotLength, szread, timeout);
         }
-        catch (IJSOCK_Exception *e) {
-            if (e->errorCode()!=JSOCKERR_timeout_expired) {
-                EXCLOG(e,"receiveDaFsBuffer(1)");
+        catch (IException *e)
+        {
+            exception.setown(e);
+            tgt.setLength(origlen);
+        }
+
+        if (exception.get())
+        {
+            if (JSOCKERR_timeout_expired == exception->errorCode())
+            {
+                EXCLOG(exception,"receiveDaFsBuffer");
                 PrintStackReport();
-                if (!tm.timemon||!tm.timemon->timedout())
-                    flushDaFsSocket(socket);
             }
-            else {
-                EXCLOG(e,"receiveDaFsBuffer");
+            else
+            {
+                EXCLOG(exception,"receiveDaFsBuffer(1)");
                 PrintStackReport();
+                if (!tm.timemon||!tm.timemon->timedout())
+                    flushDaFsSocket(socket);
             }
-            tgt.setLength(origlen);
-            throw;
+            throw exception;
```

Comment:
this needs to be:
```
    throw exception.getClear();
```

Otherwise, without either get clearing it, or linking it.The Owned<IException> will destroy it, and mean anything handling it will see a duff ptr.

Reply:
Thx, yes.